sdk/identity/identity/integration/AzureFunctions/RunTest/src/functions/authenticateToStorageFunction.ts

```diff
@@ -0,0 +1,45 @@
+
+import { BlobServiceClient } from "@azure/storage-blob";
+import { ManagedIdentityCredential } from "@azure/identity";
+import { app, HttpRequest, HttpResponseInit, InvocationContext } from "@azure/functions";
+
+export async function authenticateStorage(request: HttpRequest, context: InvocationContext): Promise<HttpResponseInit> {
+  try {
+    context.log('Http function was triggered.');
+    //parse the request body
+    await authToStorageHelper(context);
+   
+    return {
+      // status: 200, /* Defaults to 200 */
+      body: "Successfully authenticated with storage",
+    };
+  } catch (error: any) {
+    return {
+      status: 400,
+      body: error,
+    };
+  }
+};
+
+app.http('authenticateStorage', {
+    methods: ['GET', 'POST'],
+    authLevel: "anonymous",
+    handler: authenticateStorage
+});
+
+async function authToStorageHelper(context: InvocationContext): Promise<void> {
+  // This will use the system managed identity
+  const credential1 = new ManagedIdentityCredential();
+
+  const clientId = process.env.IDENTITY_USER_DEFINED_IDENTITY_CLIENT_ID!;
+  const account1 = process.env.IDENTITY_STORAGE_NAME_1;
+  const account2 = process.env.IDENTITY_STORAGE_NAME_2;
+
+  const credential2 = new ManagedIdentityCredential({"clientId": clientId });
+  const client1 = new BlobServiceClient(`https://${account1}.blob.core.windows.net`, credential1);
+  const client2 = new BlobServiceClient(`https://${account2}.blob.core.windows.net`, credential2);
+  context.log("Listing containers for storage account: system managed identity")
+  await client1.listContainers();
```

Comment:
I would make sure this is actually making the network request, otherwise it won't fail if you're unable to authenticate. Did you try [getProperties](https://learn.microsoft.com/en-us/javascript/api/@azure/storage-blob/blobserviceclient?view=azure-node-latest#@azure-storage-blob-blobserviceclient-getproperties) or [createContainer](https://learn.microsoft.com/en-us/javascript/api/@azure/storage-blob/blobserviceclient?view=azure-node-latest#@azure-storage-blob-blobserviceclient-createcontainer)?

IIRC no service call is made until you start iterating (or get `next`, etc)

Let me know if you want to chat about it!